adapters/openx/openx.go

```diff
@@ -0,0 +1,134 @@
+package openx
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+const uri = "http://rtb.openx.net/prebid"
+
+type OpenxAdapter struct {
+}
+
+type openxImpExt struct {
+	CustomParams map[string]interface{} `json:"customParams,omitempty"`
+}
+
+type openxReqExt struct {
+	DelDomain string `json:"delDomain"`
+}
+
+func (a *OpenxAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters.RequestData, []error) {
+	var errs []error
+	var validImps []openrtb.Imp
+	var reqExt openxReqExt
+
+	for _, imp := range request.Imp {
+		if err := preprocess(&imp, &reqExt); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		validImps = append(validImps, imp)
+	}
+
+	request.Imp = validImps
+	// If all the imps were malformed, don't bother making a server call with no impressions.
+	if len(request.Imp) == 0 {
+		return nil, errs
+	}
+
+	var err error
+	request.Ext, err = json.Marshal(reqExt)
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	reqJSON, err := json.Marshal(request)
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	return []*adapters.RequestData{{
+		Method:  "POST",
+		Uri:     uri,
+		Body:    reqJSON,
+		Headers: headers,
+	}}, errs
+}
+
+// Mutate the imp to get it ready to send to openx.
+func preprocess(imp *openrtb.Imp, reqExt *openxReqExt) error {
+	// We only support banner impressions for now.
+	if imp.Video != nil || imp.Native != nil || imp.Audio != nil {
```

Comment:
Small note: you may just want `if imp.Banner == nil` here? Publishers may send an `imp` with both `banner` _and_  one of these other formats, if it's a multi-format imp.